crates/client/db/src/lib.rs

```diff
@@ -305,7 +303,11 @@ impl DatabaseService {
     }
 }
 
-impl Service for DatabaseService {}
+impl Service for DatabaseService {
+    fn id(&self) -> MadaraCapability {
+        MadaraCapability::Database
```

Comment:
cancellation needs to affect the backup thread if backups are enabled

Reply:
Gotcha!

Reply:
This is actually not necessary as `spawn_backup_db_task` will run as long as ` while let Some(BackupRequest { callback, db }) = recv.blocking_recv()` is true, where `recv` is the receiver to the `backup_handle` which is stored in `MadaraBackend`, meaning the channel will automatically close when `MadaraBackend` is dropped, result in the backup thread being driven to completion. 

This is makes sense since even before we were not checking for graceful shutdown in the backup thread so there had to be another way to shut it down. Also, it would be quite complicated to get this to work with the current service architecture, as `ServiceContext` has to be retrieved from `Service::start` to be shared across services. However, `DatabaseService` does not follow this rule and starts its service on `DatabaseService::new`, which is a bit of a problem here.